design/dsl/request_test.go

```diff
@@ -0,0 +1,131 @@
+package dsl_test
+
+import (
+	"testing"
+
+	"reflect"
+
+	"goa.design/goa.v2/design"
+	. "goa.design/goa.v2/design/dsl"
+	"goa.design/goa.v2/eval"
+)
+
+func TestRequest(t *testing.T) {
+	var RequestType = Type("Request", func() {
+		Description("Optional description")
+		Attribute("required", design.String)
+		Attribute("name", design.String)
+		Required("required")
+	})
+	cases := map[string]struct {
+		DSL         func()
+		Assert      func(t *testing.T, req design.DataType)
+		ExpectError bool
+	}{
+		"DefinedUserType": {
+			DSL: func() {
+				Request(RequestType, func() {
```

Comment:
ah I know  what's wrong: the `eval.Execute` call needs to be done on the attribute expression - not the user type expression, so line 21 needs to be:
```go
	if !eval.Execute(RequestType.DSL(), RequestType.AttributeExpr) {
```
You can see the `Root` expression does the same:
https://github.com/goadesign/goa/blob/v2/design/root.go#L54